sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -598,6 +598,11 @@ class DataFrameSuite extends QueryTest {
     testData.select($"*").show(1000)
   }
 
+  test("SPARK-7319 showString") {
+    assert(testData.select($"*").showString(1).split("\n") === Seq("+---+-----+",
```

Comment:
you can use multi line string in Scala, especially with stripMargin.

http://alvinalexander.com/scala/scala-multiline-strings-heredoc-syntax